service/history/workflow/workflow_task_state_machine.go

```diff
@@ -702,9 +720,53 @@ func (m *workflowTaskStateMachine) GetWorkflowTaskInfo(
 	if scheduledEventID == workflowTask.ScheduledEventID {
 		return workflowTask, true
 	}
+
+	workflowTask = m.tryRestoreSpeculativeWorkflowTask(scheduledEventID)
+	if workflowTask != nil {
+		return workflowTask, true
+	}
+
 	return nil, false
 }
 
+func (m *workflowTaskStateMachine) tryRestoreSpeculativeWorkflowTask(
+	scheduledEventID int64,
+) *WorkflowTaskInfo {
```

Comment:
How confident are you that this is solid? We don't need this functionality for a preview release and it is always safe to add it later (we will succeed in strictly more scenarios).